File: Dubious Cyrpto.py
t = int(input())

for T in range(0, t):

    l, r, sum = map(int, input().split())
    deff = r-l
    more = True

    if sum >= l:

        for i in range(r, l-1, -1):

            if sum%i <= deff:

                a = i
                need = sum%i
                more = False
                break

            elif i-(sum%i) <= deff:

                a = i
                need = i-(sum%i)
                break

        if not more:
            print(a, l+need, l)
        else:
            print(a, l, l+need)

    else:

        need = l - sum

        print(l, l, l+need)

File: Dubious Cyrpto.cpp
#include<bits/stdc++.h>
#define ll long long int
using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t; cin>>t;

    for(int T=0; T<t; T++)
    {
        ll l,r,sum; cin>>l>>r>>sum;
        ll def = r-l;
        ll need;
        ll a;
        bool more = true;

        if(sum >= l)
        {
            for(ll i=l; i<=r; i++)
            {
                if( (sum%i) <= def)
                {
                    a = i;
                    need = sum%i;
                    more = false;
                    break;
                }
                else if( i-(sum%i) <= def)
                {
                    a = i;
                    need = i-(sum%i);
                    break;
                }
            }

            if(more == false)
            {
                cout<<a<<" "<<l+need<<" "<<l<<endl;
            }
            else{
                cout<<a<<" "<<l<<" "<<l+need<<endl;
            }
        }
        else // sum < l
        {
            need = l - sum;

            cout<<l<<" "<<l<<" "<<l+need<<endl;
        }
    }
}
